### ISSUE TITLE
Mapping network mode collapse.

### ISSUE THREAD

**Issue by GLivshits** [rosinality/stylegan2-pytorch#261]
Hi. This is not an issue but a topic for discussion. I've found out that on some domains with rich and frequent patterns there is a mode collapse of a mapping network: the generator actually uses the additive noise to generate images, but not the latents - the variation of W coordinates along the batch axis is around 1e-5.
Also, disabling noise injection at 4x4 fmap is ok, but when I disable 8x8 - there is a full mode collapse.
Is it all right that noise injection is SO important for the generation?

**Comment by rosinality** [comment]
Hmm I think the impact of noise injection is not very large if it is trained smoothly.
I think model could be trained so that noise injection have crucial role for the generation.
It may because it could be better to have spatial dimension in the latent codes. (You can may refer to https://github.com/naver-ai/StyleMapGAN)
Or it may be due to slow training rate of mapping network. You can increase learning rate of mapping network and reduce the number of layers.

**Comment by GLivshits** [comment]
I've used 0.1 lr multiplier for mapping. Note that this is not valid for all domains.
But nevertheless, the more I disable noise - the more uniform images are.
My particular domain is full of sharp details (256x256 image size) and high frequency domain. Maybe Stylegan is not capable of generating such images just using style vector.

**Comment by FinrodFelegund** [comment]
Hey, correct me if I am wrong, but these blur kernels are in place to adhere to sampling theorem before up/down sampling, right?
